raft/raft.cpp

```diff
@@ -476,8 +477,58 @@ raft::handle_heartbeat_timeout(const boost::system::error_code& ec)
     std::lock_guard<std::mutex> lock(this->raft_lock);
 
     this->request_append_entries();
+    this->notify_leader_status();
 }
 
+void
+raft::notify_leader_status()
+{
+    if(!this->enable_audit)
+    {
+        return;
+    }
+
+    audit_message msg;
+    msg.mutable_leader_status()->set_term(this->current_term);
+    msg.mutable_leader_status()->set_leader(this->uuid);
+
+    bzn::message json;
+    json["bzn-api"] = "audit";
+    json["msg"] = boost::beast::detail::base64_encode(msg.SerializeAsString());
+
+    auto json_ptr = std::make_shared<bzn::message>(json);
```

Comment:
No need to copy the stack instance of json. Just start with a shared ptr to a bzn::message and avoid this copy.